README.md

```diff
@@ -123,6 +144,92 @@ In order to cancel a job you can use :
 await deepSquareClient.cancel(jobId);
 ```
 
+## Example 
+
+Below a plain javascript fully working example launching a hello world job. 
+
+> Don't forget to setup your env
+
+
+```javascript
+import DeepSquareClient from "@deepsquare/deepsquare-client";
+import { BigNumber } from 'ethers';
+
+async function main() {
+  // Define the job
+  const helloWorldJob = {
+    "resources": {
+      "tasks": 1,
+      "gpusPerTask": 0,
+      "cpusPerTask": 1,
+      "memPerCpu": 1024
+    },
+    "enableLogging": true,
+    "steps": [
+      {
+        "name": "hello world",
+        "run": {
+          "command": "echo \"Hello World\""
+        }
+      }
+    ]
+  };
+
+  // Create the DeepSquareClient
+  const deepSquareClient = new DeepSquareClient(
+    process.env.PRIVATE_KEY as string,
+    process.env.METASCHEDULER_ADDR as string,
+    process.env.CREDIT_ADDR as string,
+    process.env.ENDPOINT as string
+  );
+
+  // Set allowance (you can also do that by loading your private key in metamask add head to https://app.deepsquare.run)
+  const depositAmount = BigNumber.from('10000000000000');
+  await deepSquareClient.setAllowance(depositAmount);
+
+  // Launch the job 
+  const randomString = Array.from({ length: 4 }, () => String.fromCharCode(65 + Math.floor(Math.random() * 26))).join('');
+  const jobId = await deepSquareClient.submitJob(helloWorldJob, `hello_world_${randomString}`);
+
+  // Print logs
+  console.log(`My job id ${jobId}`);
+  const logsMethods = deepSquareClient.getLogsMethods(jobId);
+  const [read, stopFetch] = await logsMethods.fetchLogs();
+  const decoder = new TextDecoder();
+
+  for await (const log of read) {
+    // Fetching job output here (e.g. anything printing to terminal, here Hello World)
+    console.log(decoder.decode(log.data));
+  }
+
+  stopFetch();
+}
+
+main();
+```
+
+In this modified code, the entire script is wrapped in an async function called main(). Inside the main() function, we use the await keyword to wait for the Promise returned by deepSquareClient.setAllowance(depositAmount) to resolve before continuing with the rest of the code.
+
+At the end of the script, we call main() to start the asynchronous execution of the script.
+
+
+
+
+Florin Dzeladini
+Will this wait for end of execution of main ? Or do I need to add a setTimeOut ? 
+Yes, the modified code I provided with the async function main() and main() function call at the end will wait for the end of the execution of main() before exiting the script.
+
+When you call main() at the end of the script, the async keyword indicates that it will return a Promise. However, since there is no code that explicitly awaits this Promise, Node.js will wait for the Promise to resolve or reject before exiting the script. Therefore, there is no need to add a setTimeout function.
```

Comment:
Not sure this has been left here voluntarily

Reply:
I guess not ^^